### PR TITLE
Adding Commenting and Error detection apps

### PR DESCRIPTION
**Pull request by maheshgudi**
There are 3 major additions:

1. Commenting - Created an app which fetches comments from tbc, displays it to the admin and sends notification of unread comments to the contributors.

2. Crawler - Created a Scrapy crawler which crawls through the tbc website, finds the errors on tbc notebooks, and finds the urls which return a broken/error http response.

3. Error-page/broken-page - Created an app which displays the output from the crawler to the admin.

### CODE REVIEW

File: commentingapp/commenting_new.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+        except (requests.exceptions.Timeout, requests.exceptions.ConnectionError):
+            self.internet_status = {"status": False, "message": "Please Check the internet Connection."}
+
+        return self.internet_status["message"]
+
+    def check_authentication(self, public_key, forum_name, api_version=3.0):
+
+        """ Checks if public key and forum is valid. Returns the public key, forum name for Disqus API."""
+        # @TODO - Optional Authentication for read/write/moderate.
```
Comment (ankitjavalkar) [inline commentingapp/commenting_new.py]: can stray comments be removed?

File: scripts/crawler/tbc_web_crawler/settings.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
```
Comment (ankitjavalkar) [inline scripts/crawler/tbc_web_crawler/settings.py]: Is it possible to remove the unused settings? You could also provide a link to the actual settings template in the original scrapy source code

File: scripts/split_json.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        error_json.close()
+
+    b = [saved_data for saved_data in json_data if str(saved_data).startswith("{u'br")]
+    with open(join(dirname(abspath(dirname(__file__))),'tbc_error_page/broken.json'), "w+") as broken_json:
+        cPickle.dump(b, broken_json)
+        broken_json.close()
+
+
+except ValueError:
+    print "fail"
```
Comment (ankitjavalkar) [inline scripts/split_json.py]: Maybe also print the exact error and trace back?

File: tbc_error_page/broken.json [hunk 4]
```diff
@@ -0,0 +1,130 @@
+(lp1
```
Comment (ankitjavalkar) [inline tbc_error_page/broken.json]: Do these json files need to be checked in to the repo? if not, can these be specifically added to gitignore?

File: commentingapp/commenting_new.py (hunk 1 above)
Comment (maheshgudi) [inline commentingapp/commenting_new.py]: Not yet. That's the one bit I am working upon. I am actually thinking of giving the admin the right to delete the comments from the disqus database, from the admin-commenting page itself.

File: tbc_error_page/error.json [hunk 5]
```diff
@@ -0,0 +1,194 @@
+(lp1
```
Comment (ankitjavalkar) [inline tbc_error_page/error.json]: Do these json files need to be checked in to the repo? if not can these be specifically added to gitignore?

**Comment by maheshgudi** [comment]
Not really, but its here working as a dummy. Will be removed if unecessary.

File: tbc_error_page/error.json (hunk 5 above)
Comment (maheshgudi) [inline tbc_error_page/error.json]: Not really, but its here working as a dummy. Will be removed if unecessary. Also, these are not really json files. Pickled files. My bad. Will change the names to suit the content

File: scripts/split_json.py (hunk 3 above)
Comment (maheshgudi) [inline scripts/split_json.py]: Yes, will be added to the log files. Working on it

File: scripts/crawler/tbc_web_crawler/settings.py (hunk 2 above)
Comment (maheshgudi) [inline scripts/crawler/tbc_web_crawler/settings.py]: Will be done.

File: commentingapp/commenting_new.py (hunk 1 above)
Comment (maheshgudi) [inline commentingapp/commenting_new.py]: Done

File: tbc_error_page/broken.json (hunk 4 above)
Comment (maheshgudi) [inline tbc_error_page/broken.json]: removed them

File: tbc_error_page/error.json (hunk 5 above)
Comment (maheshgudi) [inline tbc_error_page/error.json]: removed them

**Comment by ankitjavalkar** [comment]
Merged

**Pull request merged**